homeassistant/components/govee_ble/data.py

```diff
@@ -0,0 +1,263 @@
+"""Parser for Govee BLE advertisements.
+
+This file is shamlessly copied from the following repository:
+https://github.com/Ernst79/bleparser/blob/c42ae922e1abed2720c7fac993777e1bd59c0c93/package/bleparser/govee.py
+
+MIT License applies.
+"""
+from __future__ import annotations
+
+import logging
+import struct
+
+from homeassistant.components.bluetooth import BluetoothServiceInfo
+from homeassistant.components.bluetooth.device import BluetoothDeviceData
+from homeassistant.components.bluetooth.sensor import BluetoothSensorType
+from homeassistant.components.sensor import SensorStateClass
+from homeassistant.const import PERCENTAGE, TEMP_CELSIUS
+
+_LOGGER = logging.getLogger(__name__)
+
+PACKED_hHB = struct.Struct(">hHB")
+PACKED_hh = struct.Struct(">hh")
+PACKED_hhbhh = struct.Struct(">hhbhh")
+PACKED_hhhhh = struct.Struct(">hhhhh")
+
+
+NOT_GOVEE_MANUFACTURER = {76}
+
+
+def decode_temps(packet_value: int) -> float:
+    """Decode potential negative temperatures."""
+    # https://github.com/Thrilleratplay/GoveeWatcher/issues/2
+    if packet_value & 0x800000:
+        return float((packet_value ^ 0x800000) / -10000)
+    return float(packet_value / 10000)
+
+
+def decode_temps_probes(packet_value: int) -> float:
+    """Filter potential negative temperatures."""
+    if packet_value < 0:
+        return 0.0
+    return float(packet_value / 100)
+
+
+class GoveeBluetoothDeviceData(BluetoothDeviceData):
```

Comment:
I don't see how we're going to implement this in Home Assistant. Device protocol data like this is not allowed and should be in 3rd party library. Why should that use `BluetoothDeviceData`?

Reply:
Next turn will change this